### PR TITLE
Internationalization 

### PR DESCRIPTION
**Pull request by pavitrajk1989**

### CODE REVIEW

File: projects/epsilon-blueprint/src/lib/table/models/properties.ts [hunk 1]
```diff
@@ -4,6 +4,17 @@ import { Action } from './action';
...
+
+  /*Table - internationalization support starts here*/
+  selectCheckBoxLabel : string;
+  actionsVisibleHeader: string;
+  actionsHiddenHeader: string;
+  loadingText: string;
+  noDataText: string;
+  selectRowText: string;
+  actionMenuLabel: string;  
+  /*Ends here*/
```
Comment (APJohns) [inline projects/epsilon-blueprint/src/lib/table/models/properties.ts]: Let's group all the i18n key value pairs into an object in properties called internationalization

File: projects/epsilon-blueprint/src/lib/table/models/properties.ts [hunk 2]
```diff
@@ -17,8 +28,18 @@ export default interface Properties {
   actions?: Action[];
   hasColumnSelector: true;
   hasDisplayDensity: true;
+  displayDensityLabel:string;
```
Comment (APJohns) [inline projects/epsilon-blueprint/src/lib/table/models/properties.ts]: Missing space before "string"

File: projects/epsilon-blueprint/src/lib/table/models/properties.ts [hunk 3]
```diff
@@ -4,6 +4,17 @@ import { Action } from './action';
 export default interface Properties {
   caption: string;
   rowId: string;
+
+  /*Table - internationalization support starts here*/
+  selectCheckBoxLabel : string;
```
Comment (APJohns) [inline projects/epsilon-blueprint/src/lib/table/models/properties.ts]: remove space before the colon

File: projects/epsilon-blueprint/src/lib/table/table-footer/column-selector/column-selector.component.html [hunk 4]
```diff
@@ -21,7 +21,7 @@
     <ng-template ng-multi-label-tmp let-items="items" let-clear="clear">
       <div class="ng-value multiselect-value">
         <span class="ng-value-label">
-          Showing {{ selectedColumns.length }} out of {{ columnInfo.length }}
+          {{ loadingLabelText }} {{ selectedColumns.length }} {{ outOfLabelText }} {{ columnInfo.length }}
```
Comment (APJohns) [inline projects/epsilon-blueprint/src/lib/table/table-footer/column-selector/column-selector.component.html]: This doesn't solve for translating the entire string as a whole. We have an idea on how to handle this and will take a stab at it.

File: projects/epsilon-blueprint/src/lib/table/table.component.html [hunk 5]
```diff
@@ -44,14 +44,15 @@
               scope="col"
               class="th-not-sortable"
             >
-              Actions
+            {{ properties.actionsVisibleHeader }}
```
Comment (APJohns) [inline projects/epsilon-blueprint/src/lib/table/table.component.html]: Fix indentation

File: projects/epsilon-blueprint/src/lib/table/table.component.html [hunk 6]
```diff
@@ -44,14 +44,15 @@
...
             </th>
             <th
               *ngIf="properties.actions.length >= 3"
               scope="col"
               class="th-not-sortable more-actions"
             >
-              <span class="sr-only">Actions</span>
+              <span class="sr-only">
+                {{ properties.actionsHiddenHeader }}</span>
```
Comment (APJohns) [inline projects/epsilon-blueprint/src/lib/table/table.component.html]: Move `</span>` to new line

File: src/app/modules/component-library/library/table/models/properties.ts [hunk 7]
```diff
@@ -4,6 +4,17 @@ import { Action } from './action';
...
+  
+  /*Table - internationalization support starts here*/
+  selectCheckBoxLabel: string;
+  actionsVisibleHeader: string;
+  actionsHiddenHeader: string;
+  loadingText: string;
+  noDataText: string;
+  selectRowText: string;
+  actionMenuLabel: string;
+  /*Ends here*/
```
Comment (APJohns) [inline src/app/modules/component-library/library/table/models/properties.ts]: Same re: properties structure

File: projects/epsilon-blueprint/src/lib/table/models/properties.ts [hunk 8]
```diff
@@ -4,6 +4,17 @@ import { Action } from './action';
 export default interface Properties {
   caption: string;
   rowId: string;
+
```
Comment (APJohns) [inline projects/epsilon-blueprint/src/lib/table/models/properties.ts]: We should set the default values for each of these in this file

File: src/app/modules/component-library/library/table/overview/overview.component.ts [hunk 9]
```diff
@@ -25,8 +25,29 @@ export class OverviewComponent implements OnInit {
...
+      columnSelectorLabel:'Column Selector:',
+      loadingLabel:'Showing',
+      outOfLabel: 'out of',
+      defaultLabel: '(Default)',
+      displayDensityLabel: 'Display Density:',      
+      numberOfRowsLabel: 'Number of Rows per Page',
+      firstPageText: 'First Page',
+      previousPageText: 'Previous Page',
+      nextPageText: 'Next Page',
+      lastPageText: 'Last Page',
```
Comment (APJohns) [inline src/app/modules/component-library/library/table/overview/overview.component.ts]: Remove "Label" and "Text" and "Header" from key names. Instead we should make the keys reflective of the default English content
e.g.
"First Page": "First Page"

File: projects/epsilon-blueprint/src/lib/table/table-footer/column-selector/column-selector.component.html (hunk 4 above)
Comment (pavitrajk1989) [reply projects/epsilon-blueprint/src/lib/table/table-footer/column-selector/column-selector.component.html]: Sure. I was focusing on translating only text part.

File: src/app/modules/component-library/library/table/overview/overview.component.ts (hunk 9 above)
Comment (pavitrajk1989) [reply src/app/modules/component-library/library/table/overview/overview.component.ts]: I have renamed keys in align with Labels. 
E.g.
  internationalization:{
    'Select all rows': string;
    'Actions': string;
    'Loading data': string;
    'No data': string;
    'Select Row': string;
    'Actions Menu': string;
    'Display Density': string;
    'Column Selector': string;
    'Default': string;
    'Number of Rows': string;  
    'First Page': string,
    'Previous Page': string,
    'Next Page': string,
    'Last Page': string
  }

Is it good to go?

Comment (kaseyllabelle) [reply src/app/modules/component-library/library/table/overview/overview.component.ts]: @pavitrajk1989 the comments are marked resolved, but the PR hasn't been updated. Please commit and push your changes so we can re-review and then merge.

**Comment by pavitrajk1989** [comment]
Fixed code review comments

**Pull request merged**